src/index.ts

```diff
@@ -352,7 +352,8 @@ export class FigmaCalculator extends FigmaDocumentParser {
     const isLibraryComponent = (instanceNode: any) => {
       // if it's a web file, then check the componentId else the mainComponent property to get the key
       const componentKey =
-        instanceNode.componentId || instanceNode.mainComponent.key;
+        instanceNode.componentId ||
+        (instanceNode.mainComponent ? instanceNode.mainComponent.key : 0);
```

Comment:
Hey, thanks for the PR! Can you check if this syntax would also address your runtime error?

```suggestion
        instanceNode.componentId || instanceNode.mainComponent?.key;
```

Reply:
That's much nicer. Testing now.